x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -0,0 +1,310 @@
+                POPULATE_USER_METADATA
+            )
+        );
+        // Client TLS settings for incoming connections (subset of SSLConfigurationSettings)
+        set.addAll(
+            List.of(
+                CLIENT_AUTHENTICATION_TYPE,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_PATH,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_PASSWORD,
+                CLIENT_AUTHENTICATION_LEGACY_TRUSTSTORE_PASSWORD,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_ALGORITHM,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_TYPE,
+                CLIENT_AUTHENTICATION_CERT_AUTH_PATH
+            )
+        );
+        // Delegated authorization settings: authorization_realms
+        set.addAll(DelegatedAuthorizationSettings.getSettings(TYPE));
+        // JWT Cache settings
+        set.addAll(List.of(CACHE_TTL, CACHE_MAX_USERS));
+        // Standard HTTP settings for outgoing connections to get JWT issuer jwkset_path
+        set.addAll(
+            List.of(
+                HTTP_CONNECT_TIMEOUT,
+                HTTP_CONNECTION_READ_TIMEOUT,
+                HTTP_SOCKET_TIMEOUT,
+                HTTP_MAX_CONNECTIONS,
+                HTTP_MAX_ENDPOINT_CONNECTIONS,
+                HTTP_PROXY_HOST,
+                HTTP_PROXY_PORT,
+                HTTP_PROXY_SCHEME
+            )
+        );
+        // Standard TLS connection settings for outgoing connections to get JWT issuer jwkset_path
+        set.addAll(SSLConfigurationSettings.getRealmSettings(TYPE));
+        return set;
+    }
+
+    // JWT issuer settings
+
+    public static final Setting.AffixSetting<String> ALLOWED_ISSUER = RealmSettings.simpleString(
+        TYPE,
+        "allowed_issuer",
+        Setting.Property.NodeScope
+    );
+    public static final Setting.AffixSetting<TimeValue> ALLOWED_CLOCK_SKEW = Setting.affixKeySetting(
+        RealmSettings.realmSettingPrefix(TYPE),
+        "allowed_clock_skew",
+        key -> Setting.timeSetting(key, DEFAULT_ALLOWED_CLOCK_SKEW, Setting.Property.NodeScope)
+    );
+    public static final Setting.AffixSetting<List<String>> ALLOWED_SIGNATURE_ALGORITHMS = Setting.affixKeySetting(
+        RealmSettings.realmSettingPrefix(TYPE),
+        "allowed_signature_algorithms",
+        key -> Setting.listSetting(key, DEFAULT_ALLOWED_SIGNATURE_ALGORITHMS, Function.identity(), v -> {
+            if (SUPPORTED_SIGNATURE_ALGORITHMS.contains(v) == false) {
+                throw new IllegalArgumentException(
+                    "Invalid value [" + v + "] for [" + key + "]. Allowed values are " + SUPPORTED_SIGNATURE_ALGORITHMS + "}]"
+                );
+            }
+        }, Setting.Property.NodeScope)
+    );
+    public static final Setting.AffixSetting<String> JWKSET_PATH = RealmSettings.simpleString(
+        TYPE,
+        "jwkset_path",
+        Setting.Property.NodeScope
+    );
+    // Note: <allowed_issuer>.issuer_hmac_key not defined here. It goes in the Elasticsearch keystore setting.
```

Comment:
Not sure if a setting goes here for Elasticsearch keystore values.

Reply:
yes it goes here, check `RP_CLIENT_SECRET` in OIDC for an example.